zuul/connection/github.py

```diff
@@ -440,6 +455,23 @@ def getPull(self, owner, project, number):
         log_rate_limit(self.log, self.github)
         return pr
 
+    def getPullBySha(self, sha):
+        query = '%s type:pr is:open' % sha
+        pulls = set()
+        for issue in self.github.search_issues(query=query):
+            pr_url = issue.pull_request.get('url')
```

Comment:
Do all issue objects have a pull_request attribute, or do we need to check for an AttributeError here?